src/Symfony/Component/HttpKernel/CacheClearer/Psr6CacheClearer.php

```diff
@@ -20,9 +20,32 @@ class Psr6CacheClearer implements CacheClearerInterface
 {
     private $pools = array();
 
+    public function __construct(array $pools = array())
+    {
+        $this->pools = $pools;
+    }
+
     public function addPool(CacheItemPoolInterface $pool)
     {
-        $this->pools[] = $pool;
+        @trigger_error(sprintf('The %s() method is deprecated since version 3.3 and will be removed in 4.0. Pass an array of pools indexed by name to the constructor instead.', __METHOD__));
+
+        if (!in_array($pool, $this->pools, true)) {
+            $this->pools[] = $pool;
+        }
+    }
+
+    public function hasPool($name)
```

Comment:
It's convenient to use it in the command since we want to try fetching the pool from the container if the clearer can't find it. I'll first add the global clearer and come back for this point